src/rules/no-invalid-element-name.ts

```diff
@@ -60,15 +71,15 @@ const rule: Rule.RuleModule = {
             firstArg.type === 'Literal' &&
             typeof firstArg.value === 'string'
           ) {
-            const validationResult = validate(firstArg.value);
+            const validationResult = isValidElementName(firstArg.value);
             const options = context.options[0];
             const isWarning =
               !(options && options.loose) &&
-              validationResult.message !== undefined;
+              !isBestPracticeElementName(firstArg.value);
 
-            if (!validationResult.isValid || isWarning) {
+            if (!validationResult || isWarning) {
               context.report({
-                message: validationResult.message,
+                message: errorMessage,
```

Comment:
to make it a bit nicer to test, you can add the error message to the `meta`, like:

```ts
meta: {
  messages: {
    invalidName: "error goes here"
  }
}
```

then usage would be `messageId: 'invalidName'`

you don't need to import the error in the tests then and can just set `messageId` in the test expectation

Reply:
Got it, I've added and referenced it from the meta